pkg/eth/ipld_retriever.go

```diff
@@ -345,18 +347,22 @@ func (r *IPLDRetriever) RetrieveReceiptsByTxHashes(hashes []common.Hash) ([]stri
 }
 
 // RetrieveReceiptsByBlockHash returns the cids and rlp bytes for the receipts corresponding to the provided block hash
-func (r *IPLDRetriever) RetrieveReceiptsByBlockHash(hash common.Hash) ([]string, [][]byte, error) {
+func (r *IPLDRetriever) RetrieveReceiptsByBlockHash(hash common.Hash) ([]string, [][]byte, []string, error) {
 	rctResults := make([]ipldResult, 0)
 	if err := r.db.Select(&rctResults, RetrieveReceiptsByBlockHashPgStr, hash.Hex()); err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	cids := make([]string, len(rctResults))
 	rcts := make([][]byte, len(rctResults))
+	txs := make([]string, len(rctResults))
+
 	for i, res := range rctResults {
 		cids[i] = res.CID
 		rcts[i] = res.Data
+		txs[i] = res.TxHash
```

Comment:
We could do HexToHash here and return a slice of `common.Hash` instead so don't need to do it in different places above this level, but that's super nitpicky and maybe there is a reason to leave it as a hex string at this level.